src/js/web-components/video-player/VideoPlayer.js

```diff
@@ -299,4 +308,45 @@ export default class extends HTMLElement {
       assetsOnly: true,
     });
   }
+
+  /**
+   * Returns a button that controls the PiP functionality.
+   *
+   * @returns {HTMLButtonElement|null} Button element or null when PiP not supported.
+   */
+  createPiPButton() {
+    if (!('pictureInPictureEnabled' in document)) {
+      return null;
+    }
+
+    const pipButton = document.createElement('button');
+    const setPipButton = () => {
+      pipButton.disabled = (this.videoElement.readyState === 0)
+        || !document.pictureInPictureEnabled
+        || this.videoElement.disablePictureInPicture;
+    };
+
+    pipButton.innerText = 'PiP';
+    pipButton.addEventListener('click', async () => {
+      pipButton.disabled = true;
+      try {
+        if (this !== document.pictureInPictureElement) {
```

Comment:
Nit: I'd use `this.videoElement` to be explicit as it's the `<video>` sub element that is in Picture-in-Picture. +1 for consistency with `this.videoElement.requestPictureInPicture()` ;)

Reply:
@beaufortfrancois That is not going to work, actually, as `pictureInPictureElement` can't refer to an element within a `shadowRoot` tree. 

https://w3c.github.io/picture-in-picture/#documentorshadowroot-extension

The browser is supposed to run a retargeting algorithm to define a node that will be used instead. In this case it's the custom element itself.

It's an interesting caveat, I think. And it caught me by surprise, too.

Reply:
My bad, you're absolutely right. I forgot about that.